src/Protocol/ActivityPub/Receiver.php

```diff
@@ -1890,32 +1891,53 @@ private static function processAttachmentUrls(array $urls, ?string $icon): array
+				$attachments[] = ['type' => $filetype, 'mediaType' => $mediatype, 'url' => $href, 'height' => $height, 'width' => $width, 'size' => null, 'name' => '', 'image' => $icon];
 			} elseif ($filetype == 'video') {
-				$height = (int)JsonLD::fetchElement($url, 'as:height', '@value');
 				// PeerTube audio-only track
-				if ($height === 0) {
+				if (!$height) {
 					continue;
 				}
 
 				$size = (int)JsonLD::fetchElement($url, 'pt:size', '@value');
 
-				$attachments[] = ['type' => $filetype, 'mediaType' => $mediatype, 'url' => $href, 'height' => $height, 'size' => $size, 'name' => '', 'image' => $icon];
+				$attachments[] = ['type' => $filetype, 'mediaType' => $mediatype, 'url' => $href, 'height' => $height, 'width' => $width, 'size' => $size, 'name' => '', 'image' => $icon];
 			} elseif (in_array($mediatype, ['application/x-bittorrent', 'application/x-bittorrent;x-scheme-handler/magnet'])) {
-				$height = (int)JsonLD::fetchElement($url, 'as:height', '@value');
-
 				// For Torrent links we always store the highest resolution
 				if (!empty($attachments[$mediatype]['height']) && ($height < $attachments[$mediatype]['height'])) {
 					continue;
 				}
 
-				$attachments[$mediatype] = ['type' => $mediatype, 'mediaType' => $mediatype, 'url' => $href, 'height' => $height, 'size' => null, 'name' => ''];
+				$attachments[$mediatype] = ['type' => $mediatype, 'mediaType' => $mediatype, 'url' => $href, 'height' => $height, 'width' => $width, 'size' => null, 'name' => ''];
 			} elseif ($mediatype == 'application/x-mpegURL') {
 				// PeerTube uses HLS streams for video. We prefer HLS streams over the video file itself.
 				// But we still store the video file as an attachment to be used by the API which currently does not support HLS streams.
-				$attachments   = array_merge($attachments, self::processAttachmentUrls($url['as:tag'], $icon));
-				$attachments[] = ['type' => $filetype, 'mediaType' => $mediatype, 'url' => $href, 'height' => null, 'size' => null, 'name' => '', 'image' => $icon];
+				$attachments = array_merge($attachments, self::processAttachmentUrls($url['as:tag'], $icon, []));
+
+				$attachment = ['type' => $filetype, 'mediaType' => $mediatype, 'url' => $href, 'height' => $height, 'width' => $width, 'size' => null, 'name' => '', 'image' => $icon];
+				if (!empty($player)) {
+					$attachment['player-url']    = $player['embed']  ?? null;
+					$attachment['player-height'] = $player['height'] ?? null;
+					$attachment['player-width']  = $player['width']  ?? null;
+
+					foreach ($attachments as $media) {
+						if (!empty($media['height']) && !empty($media['width'])) {
+							if ($media['height'] > $attachment['player-height'] || $media['width'] > $attachment['player-width']) {
+								$attachment['player-height'] = $media['height'];
+								$attachment['player-width']  = $media['width'];
+							}
+						}
+					}
+					if (!$height && !$width) {
```

Comment:
```suggestion
					if ($height && !$width) {
```

what does JsonLD::fetchElement() return ? really "false" as boolean or "0"  or "NULL"?

Reply:
When the value isn't found, it will return `null`. But of course someone could provide `0` as value as well.

Reply:
That `if ($height && !$width) {` doesn't make sense. We need to execute the following lines when both the height and width aren't defined.

Reply:
you're right, sry